pkg/authorization/api/helpers.go

```diff
@@ -232,3 +232,47 @@ func AddUserToLSAR(user user.Info, lsar *LocalSubjectAccessReview) *LocalSubject
 	lsar.Scopes = scopes
 	return lsar
 }
+
+// +gencopy=false
+// PolicyRuleBuilder let's us attach methods.  A no-no for API types
+type PolicyRuleBuilder struct {
+	PolicyRule PolicyRule
+}
+
+func NewRule(verbs ...string) *PolicyRuleBuilder {
+	return &PolicyRuleBuilder{
+		PolicyRule: PolicyRule{
+			Verbs:         sets.NewString(verbs...),
+			Resources:     sets.String{},
+			ResourceNames: sets.String{},
+		},
+	}
+}
+
+func (r *PolicyRuleBuilder) Groups(groups ...string) *PolicyRuleBuilder {
+	r.PolicyRule.APIGroups = append(r.PolicyRule.APIGroups, groups...)
+	return r
+}
+
+func (r *PolicyRuleBuilder) Resources(resources ...string) *PolicyRuleBuilder {
+	r.PolicyRule.Resources.Insert(resources...)
+	return r
+}
+
```

Comment:
> NonResourceURLs(urls ...string)?

Not necessary for the this particular bit.  I can break the back of the problem without it.